src_c/math.c

```diff
@@ -1633,18 +1633,52 @@ vector_reflect_ip(pgVector *self, PyObject *normal)
 static double
 _vector_distance_helper(pgVector *self, PyObject *other)
 {
-    Py_ssize_t i;
-    double distance_squared, tmp;
+    Py_ssize_t i, dim = self->dim;
+    double distance_squared = 0;
+
+    /* Specialised fastpath for Vector-Vector distance calculation*/
+    if (pgVector_Check(other)) {
+        pgVector *otherv = (pgVector *)other;
+        double dx, dy;
+
+        if (dim != otherv->dim) {
```

Comment:
This should probably have used: `pgVectorCompatible_Check(otherv, dim)` ?

Reply:
I didn't use it because of 2 reasons:
- Using it would have forced me to do duplicate checks and possibly duplicate sequence obj extractions as that function doesn't tell you if the object is a vector or sequence
- It would defeat the purpose of having a fast path for vectors and one for sequences

Plus can you explain why we need to check for RealNumberCheck for each element in the sequence? PyFloat_AsDouble errors out if a complex number is passed as it can't convert to double(tested it) and so it should be enough to either check that at vector creation/assignment or not check it at all since it doesn't appear to be an issue.